hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALSplitter.java

```diff
@@ -478,6 +478,10 @@ void checkForErrors() throws IOException {
         throw new RuntimeException(thrown);
       }
     }
+
+    public Throwable clearError(){
+      return thrown.getAndSet(null);
+    }
```

Comment:
I see we make the pipelinecontroller once and use it thereafter -- it goes down into the WriterThread....  . I was going to suggest just create a new Controller on each write but looks like too much of a refactor.

Would it be cleaner if checkForErrors cleared any errors found... ; i.e. checkForErrors throws but before it throws it clears any exception in Controller?

Reply:
Not just cleaner it would also cover more usecases. I moved that line into the checkForErrors.